src/lib/genesis_ledger_helper/genesis_ledger_helper.ml

```diff
@@ -683,6 +683,41 @@ let init_from_config_file ?(genesis_dir = Cache_dir.autogen_path) ~logger
     Deferred.return
     @@ make_genesis_constants ~logger ~default:genesis_constants config
   in
+  let proof_level =
+    List.find_map_exn ~f:Fn.id
+      [ proof_level
+      ; Option.Let_syntax.(
+          let%bind proof = config.proof in
+          match%map proof.level with
+          | Full ->
+              Genesis_constants.Proof_level.Full
+          | Check ->
+              Check
+          | None ->
```

Comment:
At some point, maybe `None` should become `No_proof` or similar, to remove confusion with an option.